apps/server/src/modules/files/entity/file.entity.ts

```diff
@@ -0,0 +1,203 @@
+import { Embedded, Entity, Enum, Index, ManyToOne, Property } from '@mikro-orm/core';
+import { ObjectId } from '@mikro-orm/mongodb';
+import { v4 as uuid } from 'uuid';
+import { EntityId, StorageProviderEntity } from '@shared/domain';
+import { BaseEntityWithTimestamps } from '@shared/domain/entity/base.entity';
+import { FileOwnerModel } from '../domain';
+import { FileSecurityCheckEntity } from './file-security-check.entity';
+import { FilePermissionEntity } from './file-permission.entity';
+
+export interface FileEntityProps {
+	createdAt?: Date;
+	updatedAt?: Date;
+	deletedAt?: Date;
+	deleted?: boolean;
+	isDirectory?: boolean;
+	name: string;
+	size?: number;
+	type?: string;
+	storageFileName?: string;
+	bucket?: string;
+	storageProvider?: StorageProviderEntity;
+	thumbnail?: string;
+	thumbnailRequestToken?: string;
+	securityCheck?: FileSecurityCheckEntity;
+	shareTokens?: string[];
+	parentId?: EntityId;
+	ownerId: EntityId;
+	refOwnerModel: FileOwnerModel;
+	creatorId: EntityId;
+	permissions: FilePermissionEntity[];
+	lockId?: EntityId;
+	versionKey?: number;
+}
+
+@Entity({ collection: 'files' })
+@Index({ options: { 'permissions.refId': 1 } })
+export class FileEntity extends BaseEntityWithTimestamps {
+	@Property({ nullable: true })
+	deletedAt?: Date;
+
+	@Property()
+	deleted = false;
```

Comment:
do we need this field, or is it just duplicate information to deletedAt?

Reply:
Yes, it feels unnecessary and I think we should get rid of that in the future, but it is there now to keep the compatibility with the legacy code which operates on the `deleted` and `deletedAt` fields pair (e.g., the _mongoose-delete_ library sets both of these fields when marking the file for deletion). But basically it should be enough to have just the `deletedAt` field once me migrate all the legacy code to NestJS as the presence of it should already be a sign of the file being marked for deletion, it's not really needed to have a separate flag just for that.

Reply:
@Metauriel bump :blush: